hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/BaseHoodieClient.java

```diff
@@ -88,7 +89,14 @@ protected BaseHoodieClient(HoodieEngineContext context, HoodieWriteConfig client
   }
 
   protected BaseHoodieClient(HoodieEngineContext context, HoodieWriteConfig clientConfig,
-      Option<EmbeddedTimelineService> timelineServer) {
+                             Option<EmbeddedTimelineService> timelineServer) {
+    this(context, clientConfig, timelineServer, new TransactionManager(clientConfig, HoodieStorageUtils.getStorage(clientConfig.getBasePath(), context.getStorageConf())),
+        TimeGenerators.getTimeGenerator(clientConfig.getTimeGeneratorConfig(), context.getStorageConf()));
```

Comment:
The indentation is weird, can we fix it to be more clear.

Reply:
I'm going to move the instance construction into some methods so it reads better and is easier to read